standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/TestObjectStoreSecondaryPoolLeak.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
```

Comment:
can we move this test to a separate PR so it only contains upgrading datanucleus

Reply:
Whenever we fix a bug we usually want to have a test that proves that the bug is fixed. Why should we separate the them here?

Reply:
The main point is we use a private dependency for this test, I'm not sure if this is a good practice for the project

Reply:
There is nothing specific to `jdbc-faulty` that distinguishes it from other similar dependencies. For instance, HikariCP is maintained principally by one person, datanucleus as well, and possibly many others from those that are listed in the project.

I can understand that there is a burden when new dependencies come to the project but in the end it is a question of pros & cons. If you believe that the test does not add enough value then I am fine to remove it.